bioconda_utils/lint_functions.py

```diff
@@ -95,28 +74,35 @@ def lint_metas(recipe, metas, df, *args, **kwargs):
 
 
 @lint_multiple_metas
-def in_other_channels(recipe, meta, df):
+def in_other_channels(recipe, meta):
     """
     Does the package exist in any other non-bioconda channels?
     """
-    results = _subset_df(recipe, meta, df)
-    channels = set(results.channel).difference(['bioconda'])
-    if len(channels):
+    channels = set(utils.RepoData().get_package_data(
+        key="channel",
+        name=meta.get_value("package/name"),
+        version=meta.get_value("package/version")
+    ))
+    channels.discard('bioconda')
+    if channels:
         return {
             'exists_in_channels': channels,
             'fix': 'consider deprecating',
         }
 
 
 @lint_multiple_metas
-def already_in_bioconda(recipe, meta, df):
+def already_in_bioconda(recipe, meta):
     """
     Does the package exist in bioconda?
     """
-    results = _subset_df(recipe, meta, df)
-    build_number = int(meta.get_value('build/number', 0))
-    build_results = results[results.build_number == build_number]
-    channels = set(build_results.channel)
+    channels = utils.RepoData().get_package_data(
```

Comment:
It wouldn't make a difference. Doing `value in list` or `value in set(list)` has the same result.

In terms of performance, Python will spend most of the time figuring out which hash or comparator function to use for each item in the list. In this case, without set will be faster as it can terminate early. If the `in` was repeated often, a set would be faster. Overall though, it won't make any real difference here.